### PR TITLE
Extract locale from query strings

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Changes where we extract the locale from in query strings

From `query.plugins.i18n.locale` to `query.locale`

### Why is it needed?

To support how we are sending locales in the CM

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### CODE REVIEW

**Comment by jhoward1994** [comment]
I think this is what we need to do to fix the locale behaviour. We had changed how the locale was sent from the FE but were still looking at the wrong part of the query string in i18n.

I've tested the app and everything seems to be working but is there anywhere I could have missed?

**Review by joshuaellis — COMMENTED** [review]
I'll leave to Marc to review. But looks find from my perspective, weird how we were handling it two different ways...

File: packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ const validateLocaleCreation: Common.MiddlewareHandler = async (ctx, next) => {
     return next();
   }
 
-  const locale = get('plugins.i18n.locale', query);
+  const locale = get('locale', query);
   const relatedEntityId = get('plugins.i18n.relatedEntityId', query);
```
Comment (Marc-Roig) [inline packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts:27]: do we keep the relatedEntityId inside the plugins.i18n param too? 
Maybe we can do that in another pr

**Comment by Marc-Roig** [comment]
@joshuaellis @jhoward1994 in which endpoints we were using `locale` and not `query.plugins.i18n.locale` ? 
I believe I have always saw the latest when working on v4 endpoints

**Comment by Marc-Roig** [comment]
I don't know the full extent of where we were using this parameter, and wonder if plugins extending the admin could be using it too. I am not fully comfortable making this change in the v4 branch.

Having said this I do not see more places this could have been used.

File: packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ const validateLocaleCreation: Common.MiddlewareHandler = async (ctx, next) => {
     return next();
   }
 
-  const locale = get('plugins.i18n.locale', query);
```
Comment (Marc-Roig) [inline packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts:26]: seeing this , how come we were using the "locale" parameter in other endpoints? If only plugins.i18n.locale was read here?

**Comment by joshuaellis** [comment]
We definitely seem to use it in the `GET` routes.

File: packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts (hunk 2 above)
Comment (jhoward1994) [reply packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts:26]: Extracting the locale from the query within i18n only happens in PUT and POST endpoints (single-type/collection-type create, update, publish) see here https://github.com/strapi/strapi/blob/5a804de4144e88d598169a88403360fb8ef03da6/packages/plugins/i18n/server/src/register.ts#L24

I think this is how other CM endpoints were working okay. But in [fix/cm-publishing-locales](https://github.com/strapi/strapi/tree/fix/cm-publishing-locales) we have actually changed the query sent from the FE on PUT and POST e.g. https://github.com/strapi/strapi/blob/a6f23db9df5335e30072faf64ab2c554d88ab6a4/packages/core/admin/admin/src/content-manager/components/ContentTypeFormWrapper.tsx#L469

Which is why publishing behaviour was broken because the locale was in the wrong part of the query

wdyt the best way forward is? I think we could revert the format of PUT and POST queries on the FE if it's not okay to do this in v4

Comment (Marc-Roig) [reply packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts:26]: oh I see! If this only affects the PUT POST actions lets move forward with this, but lets be sure we are not using those parameters somewhere else !

**Review by Marc-Roig — APPROVED** [review]

**Comment by zhuanghaotao** [comment]
Hello, @jhoward1994 @Marc-Roig, in order to address some security concerns, we upgraded our Strapi version to the latest version. However, this upgrade has significantly impacted the operation of our system. We have an admin portal (similar to a content management front-end) where we publish multilingual materials through the following two POST requests:

[1] strapi-system/content-manager/collection-types/api::material.material?plugins[i18n][locale]=zh
[2] strapi-system/content-manager/collection-types/api::material.material?locale=en&relatedEntityId=3120
(zh -> default language)

The new Strapi version seems to be incorrectly creating two materials with the same language (zh) but different IDs, instead of the intended two language versions (zh and en) of the material.

We strongly advise it would be helpful to clearly document the potential risks in the developer guide document, and it'll be much appreciated to provide an additional switch or option that would allow us to choose whether to use the new query parsing approach on the backend. Otherwise, this change will require a significant amount of effort from our development team to respond to the impacts.
